GREEDY/Services/AuthenticationService.cs

```diff
@@ -0,0 +1,108 @@
+            var now = DateTime.UtcNow;
+            var tokenDescriptor = new SecurityTokenDescriptor
+            {
+                Subject = new ClaimsIdentity(new[]
+                        {
+                        new Claim(ClaimTypes.Name, username)
+                    }),
+                Expires = now.AddHours(Convert.ToInt32(expireHours)),
+
+                SigningCredentials = new SigningCredentials(new SymmetricSecurityKey(symmetricKey), SecurityAlgorithms.HmacSha256Signature)
+            };
+
+            var stoken = tokenHandler.CreateToken(tokenDescriptor);
+            var token = tokenHandler.WriteToken(stoken);
+
+            return token;
+        }
+
+        public Task<bool> ValidateToken(string token)
+        {
+            return ValidateToken(token, out string username);
+        }
+
+        public Task<bool> ValidateToken(string token, out string username)
+        {
+            username = null;
+
+            try
+            {
+                var simplePrinciple = GetPrincipal(token);
+                var identity = simplePrinciple.Identity as ClaimsIdentity;
+
+                if (identity == null)
+                {
+                    return Task.FromResult<bool>(false);
+                }
+
+                if (!identity.IsAuthenticated)
+                {
+                    return Task.FromResult<bool>(false);
+                }
+
+                var usernameClaim = identity.FindFirst(ClaimTypes.Name);
+                username = usernameClaim?.Value;
+
+                if (string.IsNullOrEmpty(username))
+                {
+                    return Task.FromResult<bool>(false);
+                }
+            }
+            catch (SecurityTokenValidationException)
+            {
+                return Task.FromResult<bool>(false);
+            }
+            
+            return Task.FromResult<bool>(true);
+        }
+
+        private ClaimsPrincipal GetPrincipal(string token)
+        {
+            try
+            {
+                var tokenHandler = new JwtSecurityTokenHandler();
+                var jwtToken = tokenHandler.ReadToken(token) as JwtSecurityToken;
+
+                if (jwtToken == null)
+                {
+                    return null;
+                }
+
+                var symmetricKey = Convert.FromBase64String(Secret);
+
+                var validationParameters = new TokenValidationParameters()
+                {
+                    RequireExpirationTime = true,
+                    ValidateIssuer = false,
+                    ValidateAudience = false,
+                    IssuerSigningKey = new SymmetricSecurityKey(symmetricKey)
+                };
+
+                var principal = tokenHandler.ValidateToken(token, validationParameters, out SecurityToken securityToken);
+
+                return principal;
+            }
+            catch (Exception e)
+            {
+                //TODO: If we have a log then we definetly need to log the exception
```

Comment:
I quest we have to have a logs. Our program getting bigger and we cannot test all parts every time. also, now we working on our personal pats, like image filtering, categorization and others and logs will be more informative for everyone and a faster issue detection during our testing time.

Reply:
I agree with the logs, but I have enough tasks on me. Maybe someone alse (like Aidas, because he doesn't know what good tasks to do) can take this one.

Reply:
sure. this was only for information. I strongly agree with suggestion to give this task for Aidas. Also, I added this task  in task board (KanbanFlow) as well